### PR TITLE
Bug 1678760 - Push Health status progress fixes

### PR DESCRIPTION
**Pull request by sarah-clements**
Screen Shot 2021-01-25 at 6 02 23 PM

The status progress indicators percentage of failures (in the pie chart) now matches the possible regressions. I also addressed a few other things - comments in the second commit or see the bug for details.

### CODE REVIEW

**Comment by ionutgoldan** [comment]
I've started looking over this, but still need some time to get familiar with the code base.

File: treeherder/webapp/api/push.py
Comment (bhearsum) [inline treeherder/webapp/api/push.py:279]: This is probably not something to address here or now, but this makes me wonder if the `Push` model should have a new status that takes into account test, build & lint, instead of re-using `testfailed`, especially since this is being done in more than one place. (I realize that this would probably be a fairly invasive change!)

File: ui/push-health/Health.jsx
Comment (bhearsum) [inline ui/push-health/Health.jsx:191]: This is a tangent, but I wonder if being able to rely on `testfailed` may be better than `needInvestigation` in my recent patch (https://github.com/mozilla/treeherder/pull/6998/files#diff-9036f1c80e93d7f3581d29707963ae23bf1427a82411009dd31e2de8913a113e) too. If so, I can update after this merges.

File: ui/shared/PushHealthStatus.jsx
Comment (bhearsum) [inline ui/shared/PushHealthStatus.jsx:103]: Am I reading this right that the solution to having the infinite spin on the Jobs view is to remove the spinner? :)

**Review by bhearsum — APPROVED** [review]
This seems pretty sensible and straightforward. I have one potential suggestion for improvement below, but it's certainly not necessary, and may even be a bad idea (I don't know the code base well enough to say).

File: treeherder/webapp/api/push.py
Comment (sarah-clements) [reply treeherder/webapp/api/push.py:279]: So do you mean showing each of those failures as separate build, lint and test failure categories in the pie chart (end consumer for this API)?

File: ui/push-health/Health.jsx
Comment (sarah-clements) [reply ui/push-health/Health.jsx:191]: That's a good point - 'needInvestigation' only refers to test failures, afaik. So if we are including lint and build failures here it'd make sense to do so consistently across the app. Especially since we'll be fixing the build and lint tabs in Push Health so it shows useful info, not just linking back to the jobs view.

File: ui/shared/PushHealthStatus.jsx
Comment (sarah-clements) [reply ui/shared/PushHealthStatus.jsx:103]: yup :) Its not necessary for when we have the data and when we don't have the data it just won't render this component.

File: treeherder/webapp/api/push.py
Comment (bhearsum) [reply treeherder/webapp/api/push.py:279]: No, I'm not suggesting any UI changes, just thinking that instead of reusing/abusing `testfailed` for counting total failures (I think right now it really is only _test_ failures?), a new field could be used instead to avoid breaking any current assumptions about it.

File: ui/push-health/Health.jsx
Comment (bhearsum) [reply ui/push-health/Health.jsx:191]: Cool! I'll wait for this to land and then rework my PR.

**Review by ionutgoldan — COMMENTED** [review]
lgtm!

File: treeherder/webapp/api/push.py
Comment (sarah-clements) [reply treeherder/webapp/api/push.py:279]: Ah, I think I see what you're saying and that does sound like a reasonable change. It'd still override `testfailed` with the `test_failure_count` so that its using the same data source for those types of failures, but I'll add separate fields for `lintfailed` and `buildfailed` and tally them in the UI.

Comment (sarah-clements) [reply treeherder/webapp/api/push.py:279]: Actually, I took another look at the code and even though its called "testfailed", the original value we're overriding does include lint and built failures. Its just being changed because of a discrepancy with the test failure count. Probably it should've been named`totalfailures` but... oh well.

Comment (bhearsum) [reply treeherder/webapp/api/push.py:279]: Hah, that's convenient!

**Pull request merged**
